### PR TITLE
[lte][agw] Add eNB and MME Status Transfer signalling

### PR DESCRIPTION
**Pull request by shaddi**
## Summary

Adds eNB and MME Status Transfer procedures for S1 mobility.

Signed-off-by: Shaddi Hasan <shasan@fb.com>

## Test Plan

- Local testing (2x Baicells eNB + Pixel 3)
- TeraVM + Spirent

### CODE REVIEW

File: lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c [hunk 1]
```diff
@@ -2565,6 +2565,110 @@ int s1ap_mme_handle_handover_command(
...
+      target_enb_association =
+          (enb_description_t*) (uintptr_t) enb_array->elements[idx];
+      if (target_enb_association->enb_id ==
+          ue_ref_p->s1ap_handover_state.target_enb_id) {
+        break;
+      }
+    }
+    free_wrapper((void**) &enb_array->elements);
+    free_wrapper((void**) &enb_array);
+    if (target_enb_association->enb_id !=
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c:2663]: null check?

File: lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c [hunk 2]
```diff
@@ -2565,6 +2565,110 @@ int s1ap_mme_handle_handover_command(
...
+        ue_ref_p->s1ap_handover_state.target_enb_id);
+    OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+  }
+
+  bstring b = blk2bstr(buffer, length);
+  free(buffer);
+
+  s1ap_mme_itti_send_sctp_request(
+      &b, target_enb_association->sctp_assoc_id, 0x0,
+      ue_ref_p->mme_ue_s1ap_id);  // replace 0x0 with target ue stream
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c]: looks like you did forget to do what the comment on this line is stating?

Comment (shaddi) [reply lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c]: YIkes! Good catch...

File: lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c (hunk 1 above)
Comment (ulaskozat) [reply lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c:2663]: my comment on null check looks redundant, the check on line 2628 should be sufficient...

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**